### PR TITLE
Add a rake task to profile basic memory usage

### PR DESCRIPTION
**Pull request by ashmaroli**
To use Travis to constantly gauge allocations from a basic usage scenario

### CODE REVIEW

**Comment by pyrmont** [comment]
I like the idea of adding tooling to improve development but I don't know if this is something that should be run on every CI build.

**Comment by ashmaroli** [comment]
> I don't know if this is something that should be run on every CI build.

Currently, the task profiles just a single lexer and formatter. Eventually, the task could be upgraded to profile the entire test-suite. That would should how much memory gets used from most known scenarios cumulatively.
By having it run on every build, those watching this project *closely* can gauge the effect of every commit to `master` on memory consumption.

File: Gemfile [hunk 1]
```diff
@@ -15,6 +15,11 @@ gem 'rubocop', '~> 0.49.1'
 # don't try to install redcarpet under jruby
 gem 'redcarpet', :platforms => :ruby
 
+# Profiling
```
Comment (pyrmont) [inline Gemfile]: Looking at the Gemfile, I wonder if we should create a :test group for some of these gems that we want to use in development and testing but that aren't necessary for production.

Something like:

```ruby
# frozen_string_literal: true

source 'http://rubygems.org'

gemspec

gem 'rake'

gem 'redcarpet', :platforms => :ruby # don't try to install redcarpet under jruby

group :development, :test
  # automated testing
  gem 'minitest', '>= 5.0'
  gem 'minitest-power_assert'
  gem 'rubocop', '~> 0.49.1'

  # benchmarking
  gem 'memory_profiler', :require => false if RUBY_VERSION >= '2.3.0' 

  # make CI builds faster
  gem 'parallel', '~> 1.13.0' if RUBY_VERSION < '2.2.0'
end

group :development do
  gem 'pry'

  # docs
  gem 'yard'
  gem 'github-markup'

  # visual tests
  gem 'shotgun'
  if RUBY_VERSION < '2.2.0'
    gem 'sinatra', '~> 1.4.8' 
  else
    gem 'sinatra'
  end
end

**Comment by jneen** [comment]
One nice thing about rouge is that it has no actual dependencies - everything in the Gemfile is for development purposes. In production one would expect rouge to be used as a gem, and for that the `gemspec` is what matters.

File: Gemfile (hunk 1 above)
Comment (ashmaroli) [reply Gemfile]: I think having a `:test` group is good for organizing the gems and limit the gems installed in CI builds (faster *install* phase, lesser gems to cached between builds).

But doing that is not within the scope of this PR. Once the `:test` group has been added to the `master` branch, I'll update this PR..

Comment (pyrmont) [reply Gemfile]: As discussed in 1146, we decided to stick with the original approach.

**Pull request merged**